From 5d0f5202f87dcbf9072f5219dfc5005e072dbf27 Mon Sep 17 00:00:00 2001
From: Nathan Mittler <nmittler@gmail.com>
Date: Mon, 9 May 2022 11:34:52 -0700
Subject: [tf] Fix deployment of VMs (#38815)

It's not clear why installation succeeds on prow, but without this change the tests will not run on GKE.
---
 pkg/test/framework/components/cluster/kube/cluster.go | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/pkg/test/framework/components/cluster/kube/cluster.go b/pkg/test/framework/components/cluster/kube/cluster.go
index e80af8e0c5..efabfa0baf 100644
--- a/pkg/test/framework/components/cluster/kube/cluster.go
+++ b/pkg/test/framework/components/cluster/kube/cluster.go
@@ -43,12 +43,17 @@ type Cluster struct {
 // CanDeploy for a kube cluster returns true if the config is a non-vm, or if the cluster supports
 // fake pod-based VMs.
 func (c *Cluster) CanDeploy(config echo.Config) (echo.Config, bool) {
-	if config.DeployAsVM && !c.vmSupport {
+	if config.DeployAsVM && !c.isVMSupported() {
 		return echo.Config{}, false
 	}
 	return config, true
 }
 
+func (c *Cluster) isVMSupported() bool {
+	// VMs can only be deployed on config clusters, since they assume the cluster ID of the control plane.
+	return c.IsConfig() && c.vmSupport
+}
+
 // OverrideTopology allows customizing the relationship between this and other clusters
 // for a single suite. This practice is discouraged, and separate test jobs should be created
 // on a per-topology bassis.
-- 
2.35.3

